### ISSUE TITLE
fans not stopping on a cards supporting 0 rpm

### ISSUE THREAD

**Issue by Rootax** [Rem0o/FanControl.Releases#265]
Hello there.

Congrats for this incredible project, really.

I've only 1 problem with it.

I've a MSI 3090 which supports 0 rpms. When temps are low, the fans stop. When I'm using Fan Control to .. control them, even if the curb says 0%, the fan never go under 30% speed. If I untick them in Fan Control, It goes to o rpm allright, but after a while, even without load, it's going to 30% again. Like if Fan Control doesn't like the fact that fans are reporting 0%.

Is this fixable ? Thx :)

Windows 10 x64 , Fan Control V83.

**Comment by Korkman** [comment]
Do you have MSI Afterburner installed as well? Make sure to disable "user defined software automatic fan control" (custom fan curve) in settings, or the two will interfer.

**Comment by Rootax** [comment]
I've MSI afterburner yes, but I'm not using it to control fan curve. I checked again and user defined software automatic fan control is not enabled.

**Comment by Rem0o** [comment]
The driver implementation I'm using to control the card's fan doesn't seem to like anything below 30% for RTX 30xx cards.
Currently using https://github.com/falahati/NvAPIWrapper for that.

**Comment by Rootax** [comment]
And no new commit since December, the project look dead...  :/

**Comment by Rem0o** [comment]
AFAIK, Nvidia didn't even update their API documentation since 2019 https://docs.nvidia.com/gameworks/content/gameworkslibrary/coresdk/nvapi/index.html

**Comment by Korkman** [comment]
It works for me. ASUS Strix 3080 OC. When I set it to zero RPM in Fan Control, it obeys. I can even overheat it, so temperature doesn't seem to trigger a minimum RPM of any kind.

**Comment by Rootax** [comment]
But does it come back to 0 after the fans spin up under load ?

**Comment by Korkman** [comment]
@Rootax Yes. I can set a curve including zero rpm, put it under load and idle again, behaves as it should. Do you have two GPU fan groups? I noticed that with my new card, I have two fan controls for Nvidia, controlling two plus one fan.

**Comment by Rootax** [comment]
Yeah 2 fan groups too.

If I set them to 0% it will stop but not for long, and it's not based on
temps. More like something reloads after or... I don't know.

With that said some users of other softwares are saying that yeah, no fan
speed under 30% for the 30 series, so it's not a problem directly related
to Fan Control.

**Comment by S1332** [comment]
I have a Ventus 3x, and weirdly it works when everything is set of, but it seems that the card itself is blocking 0% as soon as either afterburner or Fancontrol interfere with fan speeds.

**Comment by 0bsidian** [comment]
I also have the same issue with an nVidia RTX 3060 Ti FE. Even if I set a flat 0% fan curve, both fan controls for the GPU are locked at 30%.

**Comment by BouleJaune** [comment]
I have the same issue with a 3080ti. But even if I don't control the GPU's fans they won't stop (39°c). Is there at least a way to not hinder the normal behavior when not controlling them ?

**Comment by Y4m4k** [comment]
I had the same issue with a 3070 Ti. Turns out that if you mess with the Nvidia Control Panel setting, it will impact the wattage of the GPU and hence the fan speeds. See this link: https://github.com/Rem0o/FanControl.Releases/discussions/607

**Comment by rasto-pa** [comment]
As a hint in MSI Afterburner says that some GPUs don't allow full manual fan speed adjustment. It has checkbox "Override zero fan speed" that enables 0% speed with manual curve.
@Rem0o Is it possible to implement it in FanControl?

**Comment by Rem0o** [comment]
@rasto-pa  if the information on how they did it, sure.  But in any case it seems like kind of a hack on their part, there is no official support in the API to do so as of now.

**Comment by peterorosi** [comment]
Just bought MSI 3080 Suprim X 10G which has the same problem but in addition to that the "Override zero fan speed" in Afterburner doesn't seem to work either. On the surface this may not seem as too serious a problem but to have 30% = 900 rpm at idle in otherwise silent build is basically a dealbreaker for me (as is the alternative to have the fan curve controlled directly by firmware, which supports fan stop, but the full load noise handling is just terrible)

But I have a suggestion for workaround that should be relatively easy to implement and I don't see reason why it shouldn't work. The only reason why some 3000 cards have fan stop problems in "Fan Control" app is that once the rpm gets above 30% the fan control can't reduce the rpm below 30% in a standard way using API calls (it can only move within 30% - 100% window) However if I disable the control (manually clicking on the toggle) and then immediately reanable it, then the the fan rpm resets to 0% as it should and stays there as long as the target rpm remains below 30%. So in theory all that I would need when switching i.e. from gaming to desktop would be to manually disable and then reanable the fan control and all would be fine. However this is super annoying and unacceptable to do manually.

Instead I suggest that this should be done autmatically by the SW by introducing some "Reset control threshold" and anytime the target rpm % would flip from values above the threshold % into values below the threshold % then reinitialization of the control would happen (equivalent to when user manually disables it and the subsequently enables it.)

When goingo from full load to idle, there is typically only one occurence when this reinitialization needs to be done (i.e. first occurence of target rpm % that is < 30% assuming the threshold is 30%) - so performancewise it shouldn't be a problem because after the reset is done it doesn't need to be repeated again until the target rmp doesn't go above 30% (i.e. new intense task is run on the GPU) and only after this intense task finishes and we cross the threshould once again (from above 30% -> below 30%)

The "Reset control threshold" or "Minimum rpm % threshold" could be a configurable numeric parameter although if the goal would be to fix Ampere cards specifically, then it could be hardcoded to 30% and on UI it would be enough to provide only single checkbox "Hack Ampere min rpm" or similar.

Please let me know if this hack could work or alternatively if I should start looking into how to get rid of my new 3080...
